clustered/client/src/main/java/org/ehcache/clustered/client/internal/EhcacheClientEntity.java

```diff
@@ -90,7 +90,7 @@
   private final LifeCycleMessageFactory messageFactory;
   private final Map<Class<? extends EhcacheEntityResponse>, List<ResponseListener<? extends EhcacheEntityResponse>>> responseListeners = new ConcurrentHashMap<Class<? extends EhcacheEntityResponse>, List<ResponseListener<? extends EhcacheEntityResponse>>>();
   private final List<DisconnectionListener> disconnectionListeners = new CopyOnWriteArrayList<DisconnectionListener>();
-  private final List<ReconnectListener> reconnectListeners = new ArrayList<ReconnectListener>();
+  private final List<ReconnectListener> reconnectListeners = new CopyOnWriteArrayList<ReconnectListener>();
```

Comment:
I'm still not sure if the old way of using the `ArrayList` and synchronizing every access is better over the use of this `CopyOnWriteArrayList`. Did that anyway so that it's consistent with the other listeners. Is there gonna any significant performance difference? Which one is preferred?